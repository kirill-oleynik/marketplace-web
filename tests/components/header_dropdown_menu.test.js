const React = require('react');
const { shallow } = require('enzyme');
const toJSON = require('enzyme-to-json').default;
const {
  HeaderDropdownMenu
} = require('../../app/components/header_dropdown_menu');

describe('#render', () => {
  test('it renders correctly', () => {
    const component = shallow(
      <HeaderDropdownMenu
        t={(translation) => translation}
        currentUser={{}}
        openProfile={() => {}}
      />
    );

    expect(
      toJSON(component)
    ).toMatchSnapshot();
  });
});

describe('#toggle', () => {
  test('it changes dropdownOpen state', () => {
    const component = shallow(
      <HeaderDropdownMenu
        t={(translation) => translation}
        currentUser={{}}
        openProfile={() => {}}
      />
    );

    expect(component.state().dropdownOpen).toEqual(false);

    component.instance().toggle();

    expect(component.state().dropdownOpen).toEqual(true);
  });
});
