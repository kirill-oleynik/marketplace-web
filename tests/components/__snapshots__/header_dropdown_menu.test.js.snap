// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#render it renders correctly 1`] = `
<Dropdown
  className="main-dropdown header-dropdown"
  isOpen={false}
  tag="div"
  toggle={[Function]}
>
  <DropdownToggle
    aria-haspopup={true}
    caret={true}
    className="main-dropdown__toggle header-dropdown__btn"
    color="secondary"
    data-toggle="dropdown"
  />
  <DropdownMenu
    className="header-dropdown__menu"
    right={true}
    tag="div"
  >
    <div
      className="header-dropdown__menu-wrap"
    >
      <p
        className="font-14 font-700 mb-5"
      />
      <p
        className="font-12 in-black-050"
      />
    </div>
    <div
      className="divider divider--dark"
    />
    <div
      className="header-dropdown__menu-wrap"
    >
      <button
        className="font-14 mb-20"
        onClick={[Function]}
      >
        header.viewProfile
      </button>
      <MainButton
        className="w-100 mb-15"
        color="blue"
        size="sm"
        type="button"
      >
        header.submitApp
      </MainButton>
    </div>
    <div
      className="divider divider--dark"
    />
    <div
      className="header-dropdown__menu-wrap mb-10"
    >
      <span
        className="header-modal__link header-modal__link--dark"
        onClick={[Function]}
      >
        <i
          className="header-modal__logout icon icon-log-out in-black-035"
        />
        <span
          className="d-inline-block align-middle"
        >
          header.logOut
        </span>
      </span>
    </div>
  </DropdownMenu>
</Dropdown>
`;
